log_collector.py

```diff
@@ -18,4 +18,7 @@ def get_log(api_endpoint, admin, access_key, username, job_id, write=False):
     if write:
         with open(log_name, 'w') as log:
             log.write(resp.text)
+    if resp.status_code == 404:
+        print("404 API response.  The asset is no longer there (> 30 days since test creation) or does not exist.")
+        return None
```

Comment:
Can we, instead of printing, create and throw a custom exception?
Something like this:
https://github.com/enriquegh/testobject-python-api/blob/230a3db643b314ddcfb45910c13af8520f913fe3/testobject/client.py#L14

Reply:
Not familiar with custom exceptions @enriquegh, can I add an exception and print from inside it?  Then catch it and return None?

Is it "pythonic" to catch an exception then return None from the function?  I need to read more about exceptions, it seems like in Python they're not meant to be equivalent to Errors.  They're more like control flow components and you can throw an exception even if the situation is not "Exceptional".

Reply:
I tried a few things and I can't get it to work.

Unsure if I'm misunderstanding how Exceptions work in Python or if I need to restructure how `fetch_log` and `get_log` function.  But if I throw an exception instead of returning None then the error can't bubble up to __main__

I think the main problem is I'm relying on `None` here: https://github.com/mdsauce/sauce-log-parser/blob/60ddd4b64fe4388210002e139163baf4006b22b4/sel_log_parser.py#L127

If an Exception is introduced in lieu of `None` then the for loop on line 118 doesn't have a way to `continue`.

Reply:
discussed this in person.  will be making a commit for #25 to refactor this loop.